codec-http/src/test/java/io/netty/handler/codec/http/websocketx/WebSocketCloseStatusTest.java

```diff
@@ -0,0 +1,129 @@
+        PROTOCOL_ERROR,
+        INVALID_MESSAGE_TYPE,
+        INVALID_PAYLOAD_DATA,
+        POLICY_VIOLATION,
+        MESSAGE_TOO_BIG,
+        MANDATORY_EXTENSION,
+        INTERNAL_SERVER_ERROR
+    );
+
+    @Test
+    public void testKnownStatuses() {
+        Assert.assertSame(NORMAL_CLOSURE, valueOf(1000));
+        Assert.assertSame(ENDPOINT_UNAVAILABLE, valueOf(1001));
+        Assert.assertSame(PROTOCOL_ERROR, valueOf(1002));
+        Assert.assertSame(INVALID_MESSAGE_TYPE, valueOf(1003));
+        Assert.assertSame(INVALID_PAYLOAD_DATA, valueOf(1007));
+        Assert.assertSame(POLICY_VIOLATION, valueOf(1008));
+        Assert.assertSame(MESSAGE_TOO_BIG, valueOf(1009));
+        Assert.assertSame(MANDATORY_EXTENSION, valueOf(1010));
+        Assert.assertSame(INTERNAL_SERVER_ERROR, valueOf(1011));
+    }
+
+    @Test
+    public void testUserDefinedStatuses() {
+        // Given
+        WebSocketCloseStatus feedTimeot = valueOf(6033, "Feed timed out");
+        WebSocketCloseStatus untradablePrice = valueOf(6034, "Untradable price");
+
+        // When
+        WebSocketCloseStatus status_6033 = valueOf(6033);
+        WebSocketCloseStatus status_6034 = valueOf(6034);
+
+        // Then
+        Assert.assertSame(feedTimeot, status_6033);
+        Assert.assertEquals(feedTimeot.code(), 6033);
+        Assert.assertEquals(feedTimeot.reasonText(), "Feed timed out");
+
+        Assert.assertSame(untradablePrice, status_6034);
+        Assert.assertEquals(untradablePrice.code(), 6034);
+        Assert.assertEquals(untradablePrice.reasonText(), "Untradable price");
+    }
+
+    @Test
+    public void testUserLaterDefinedStatuses() {
+        // Given
+        int statusCode = 6035;
+        String message = "Out of liquidity";
+
+        // When
+        WebSocketCloseStatus outOfLiquidity1 = valueOf(statusCode);
+
+        // Then
+        Assert.assertSame(outOfLiquidity1, valueOf(statusCode));
+        Assert.assertEquals(outOfLiquidity1.code(), statusCode);
+        Assert.assertEquals(outOfLiquidity1.reasonText(), "Close status #" + statusCode);
+
+        // When
+        WebSocketCloseStatus outOfLiquidity2 = valueOf(statusCode, message);
+
+        // Then
+        Assert.assertSame(outOfLiquidity1, outOfLiquidity2);
+        Assert.assertEquals(outOfLiquidity1.code(), statusCode);
+        Assert.assertEquals(outOfLiquidity1.reasonText(), message);
+    }
+
+    @Test
+    public void testRfc6455CodeValidation() {
+        // Given
+        List<Integer> knownCodes = Arrays.asList(
+            NORMAL_CLOSURE.code(),
+            ENDPOINT_UNAVAILABLE.code(),
+            PROTOCOL_ERROR.code(),
+            INVALID_MESSAGE_TYPE.code(),
+            INVALID_PAYLOAD_DATA.code(),
+            POLICY_VIOLATION.code(),
+            MESSAGE_TOO_BIG.code(),
+            MANDATORY_EXTENSION.code(),
+            INTERNAL_SERVER_ERROR.code()
+        );
+
+        SortedSet<Integer> invalidCodes = new TreeSet<Integer>();
```

Comment:
`Set`

Reply:
'Set' does NOT have '.first()' and '.last()' methods used by this test. SortedSet is used intentionally.

Reply:
Ah sorry mist this.